src/microcosm.js

```diff
@@ -121,23 +116,40 @@ inherit(Microcosm, Emitter, {
 
     this.updateSnapshot(action, next)
 
-    return this
+    this.state = next
   },
 
-  prepareRelease () {
-    let next = this.recall(this.history.head)
+  release (action) {
+    this.changes.update(this.state)
+    this.effects.dispatch(action)
+  },
 
-    this.dirty = next !== this.state
-    this.state = next
+  on (type, callback, scope) {
+    let [event, meta] = type.split(':', 2)
+
+    switch (event) {
+      case 'change':
+        this.changes.on(meta || '', callback, scope)
+        break;
+      default:
+        Emitter.prototype.on.apply(this, arguments)
+    }
+
+    return this
   },
 
-  release (action) {
-    if (this.dirty) {
-      this.dirty = false
-      this._emit('change', this.state)
+  off (type, callback, scope) {
+    let [event, meta] = type.split(':', 2)
+
+    switch (event) {
+      case 'change':
+        this.changes.off(meta, callback, scope)
+        break;
+      default:
+        Emitter.prototype.off.apply(this, arguments)
     }
 
-    this.effects.dispatch(action)
+    return this
```

Comment:
Switching like this is for backwards compatibility. I don't want to break apps listening to `on('teardown')`, even though I don't believe it is documented.